scipy-stubs/interpolate/_bsplines.pyi

```diff
@@ -11,66 +11,109 @@ __all__ = ["BSpline", "make_interp_spline", "make_lsq_spline", "make_smoothing_s
 
 ###
 
+_CT_co = TypeVar("_CT_co", bound=np.float64 | np.complex128, default=np.float64, covariant=True)
```

Comment:
does this exclude float32? If yes, `BSpline.__init__` accepts it (and upcasts to float64; ditto for complex64 -> complex128); same for `PPoly.__init__` and `BPoly.__init__` (this comment applies everywhere unless noted explicitly)

Reply:
`__init__` still accepts any array-like that can be converted to `float64` or `complex128`. The constructor functions and methods use `@overload` to "select" either `float64` or `complex`, based on whether the input is "float-like" or "complex-like". If it can't be determined (e.g. when passing an array type of `np.inexact`), it'll fall back to `Any`.